test/features/public_asset/domain/extensions/string_to_public_asset_extension_test.dart

```diff
@@ -5,17 +5,50 @@ import 'package:tmail_ui_user/features/public_asset/domain/extensions/string_to_
 void main() {
   group('string to public asset extension test:', () {
     test(
-      'should convert string to public assets',
+      'should convert string to public assets '
+      'when image tags have public-asset-id attribute',
     () {
       // arrange
-      const publicAssetIdValue = 'public-asset-1';
-      const htmlString = '<img src="public-asset-1" public-asset-id="$publicAssetIdValue"/>';
+      const publicAssetIdValue1 = 'public-asset-1';
+      const publicAssetIdValue2 = 'public-asset-2';
+      const htmlString = '<img src="public-asset-1" public-asset-id="$publicAssetIdValue1"/>'
+        '<img src="public-asset-2" public-asset-id="$publicAssetIdValue2"/>';
 
       // act
       final publicAssetIds = htmlString.publicAssetIdsFromHtmlContent;
       
       // assert
-      expect(publicAssetIds, equals([Id(publicAssetIdValue)]));
+      expect(publicAssetIds, equals([Id(publicAssetIdValue1), Id(publicAssetIdValue2)]));
+    });
+
+    test(
+      'should convert string to nothing '
+      'when image tags do not have public-asset-id attribute',
+    () {
+      // arrange
+      const htmlString = '<img src="public-asset-1"/><img src="public-asset-2"/>';
+
+      // act
+      final publicAssetIds = htmlString.publicAssetIdsFromHtmlContent;
+      
+      // assert
+      expect(publicAssetIds, isEmpty);
+    });
+
+    test(
+      'should only convert string to public assets '
+      'when image tags have public-asset-id attribute',
+    () {
+      // arrange
+      const publicAssetIdValue1 = 'public-asset-1';
+      const htmlString = '<img src="public-asset-1" public-asset-id="$publicAssetIdValue1"/>'
+        '<img src="public-asset-2"/>';
+
+      // act
+      final publicAssetIds = htmlString.publicAssetIdsFromHtmlContent;
+      
+      // assert
+      expect(publicAssetIds, equals([Id(publicAssetIdValue1)]));
```

Comment:
test more with cases:
- `<img src="public-asset-1" public-asset-id="id1" public-asset-id="id2"/>`
- `<img src="public-asset-id" public-asset-id="id1"/>`
- `<img src="https://domain.com/public-asset-id" public-asset-id="id1"/>`
- `<img src="public-asset-1" public-asset-id="#$52323"/>`
- `<img src="public-asset-1" public-at-id="id1" public-asset-id="id2"/>`

Reply:
>- `<img src="public-asset-id" public-asset-id="id1"/>`
>- `<img src="https://domain.com/public-asset-id" public-asset-id="id1"/>`

This extension doesn't care about `src` attribute, so these will not be tested further.

Reply:
Done